### PR TITLE
IDE-3992 make gradle and maven project creating backstage

### PR DESCRIPTION
**Pull request by Camork**
make module project and workspace project creating in below steps:
1.open project
2.import opened project
3.build project

### CODE REVIEW

File: maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/MavenUtil.java [hunk 1]
```diff
@@ -595,6 +645,14 @@ public static void setConfigValue(Xpp3Dom configuration, String childName, Objec
 		childNode.setValue((value == null) ? null : value.toString());
 	}
 
+	public static void setMavenNature(IProject project) throws CoreException {
+		IProjectDescription description = project.getDescription();
+
+		description.setNatureIds(new String[] {"org.eclipse.m2e.core.maven2Nature"});
```
Comment (jtydhr88) [inline maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/MavenUtil.java]: will it replace the existing natures?

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java [hunk 2]
```diff
@@ -135,6 +131,8 @@ public void stop(BundleContext context) throws Exception {
 		super.stop(context);
 	}
 
+	// The shared instance
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java]: if it is no useful, just remove it

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java [hunk 3]
```diff
@@ -115,6 +118,22 @@ public static IStatus createErrorStatus(String msg) {
 		return createErrorStatus(PLUGIN_ID, msg);
 	}
 
+	public static IStatus createErrorStatus(String msg, boolean erroLog) {
+		if (erroLog) {
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java]: why need this one?

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java [hunk 4]
```diff
@@ -131,14 +150,6 @@ public static IStatus createWarningStatus(String message) {
 		return new Status(IStatus.WARNING, PLUGIN_ID, message);
 	}
 
-	public static IStatus createWarningStatus(String message, String id) {
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java]: why remove those two methods?

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/ImportLiferayWorkspaceOpMethods.java [hunk 5]
```diff
@@ -54,51 +53,42 @@ public static Status execute(ImportLiferayWorkspaceOp op, ProgressMonitor pm) {
...
 
-			retval = StatusBridge.create(status);
+			LiferayWorkspaceUtil.clearWorkspace(location);
 
-			if (retval.ok()) {
-				Value<Boolean> provisionLiferayBundle = op.getProvisionLiferayBundle();
+			IPath eclipseWsLocation = PathBridge.create(wsLocation);
 
-				boolean initBundle = provisionLiferayBundle.content();
+			IStatus importStatus = provider.importProject(eclipseWsLocation,wsName, monitor);
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/ImportLiferayWorkspaceOpMethods.java]: sf

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/NewLiferayWorkspaceProjectProvider.java [hunk 6]
```diff
@@ -27,8 +28,8 @@
 
 	public String getInitBundleUrl(String workspaceLocation);
 
-	public IStatus importProject(String location, IProgressMonitor monitor);
+	public IStatus importProject(IPath wsLocation, String wsName, IProgressMonitor monitor);
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/NewLiferayWorkspaceProjectProvider.java]: you just get workspaceName from 
String wsName = wsLocation.lastSegment(); in previous code, why must the String wsName here?

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleWorkspaceProjectProvider.java [hunk 7]
```diff
@@ -112,49 +129,80 @@ public String getInitBundleUrl(String workspaceLocation) {
...
 	@Override
-	public void initBundle(String bundleUrl, String serverName, String workspaceName, IProgressMonitor monitor) {
-		IProject project = ProjectUtil.getProject(workspaceName);
+	public void initBundle(String bundleUrl, String serverName, String wsName) {
+		IWorkspaceRoot workspaceRoot = CoreUtil.getWorkspaceRoot();
 
-		try {
-			if (bundleUrl != null) {
-				IFile gradlePropertiesFile = project.getFile("gradle.properties");
+		IProject project = workspaceRoot.getProject(wsName);
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleWorkspaceProjectProvider.java]: just call
IProject project = ProjectUtil.getProject(workspaceName);

File: ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java [hunk 8]
```diff
@@ -34,7 +34,7 @@ public void createActivator() {
 
 		wizardAction.newModule.prepareGradle(project.getName(), ACTIVATOR);
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java [hunk 9]
```diff
@@ -47,7 +47,7 @@ public void createApi() {
 
 		wizardAction.newModule.prepareGradle(project.getName(), API);
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java [hunk 10]
```diff
@@ -60,7 +60,7 @@ public void createContentTargetingReport() {
 
 		wizardAction.newModule.prepareGradle(project.getName(), CONTENT_TARGETING_REPORT);
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java [hunk 11]
```diff
@@ -73,7 +73,7 @@ public void createContentTargetingRule() {
 
 		wizardAction.newModule.prepareGradle(project.getName(), CONTENT_TARGETING_RULE);
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java [hunk 12]
```diff
@@ -86,7 +86,7 @@ public void createContentTargetingTrackingAction() {
 
 		wizardAction.newModule.prepareGradle(project.getName(), CONTENT_TARGETING_TRACKING_ACTION);
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.module/com.liferay.ide.ui.module.wizard/com.liferay.ide.ui.module.wizard.create/src/com/liferay/ide/ui/module/tests/NewModuleProjectWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/tools/com.liferay.ide.ui.workspace/com.liferay.ide.ui.workspace.wizard/com.liferay.ide.ui.workspace.wizard.create.tomcat.70/src/com/liferay/ide/ui/workspace/tests/NewLiferayWorkspaceWizardGradleTests.java [hunk 13]
```diff
@@ -36,7 +35,7 @@ public void createLiferayWorkspace() {
 
 		wizardAction.newLiferayWorkspace.prepareGradle(project.getName());
 
-		wizardAction.finish();
+		wizardAction.finishWithNoRuningJobs();
```
Comment (jtydhr88) [inline ui-tests/tools/com.liferay.ide.ui.workspace/com.liferay.ide.ui.workspace.wizard/com.liferay.ide.ui.workspace.wizard.create.tomcat.70/src/com/liferay/ide/ui/workspace/tests/NewLiferayWorkspaceWizardGradleTests.java]: lets call
wizardAction.finish();
jobAction.waitForNoRunningJobs();

File: ui-tests/common/com.liferay.ide.ui.liferay/src/com/liferay/ide/ui/liferay/action/WizardAction.java [hunk 14]
```diff
@@ -84,6 +84,12 @@ public void finish() {
 		ide.sleepLinux(2000);
 	}
 
+	public void finishWithNoRuningJobs() {
```
Comment (jtydhr88) [inline ui-tests/common/com.liferay.ide.ui.liferay/src/com/liferay/ide/ui/liferay/action/WizardAction.java]: do not add this method,
just call jobAction.waitForNoRunningJobs()

**Comment by jtydhr88** [comment]
see comments and please ask some qa help to test on Linux and Mac to make sure your code works well on all os, thanks.

File: maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/MavenUtil.java (hunk 1 above)
Comment (Camork) [reply maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/MavenUtil.java]: Yes, i miss it.

Comment (jtydhr88) [reply maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/MavenUtil.java]: please check the m2e have the method to do this or not

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java (hunk 3 above)
Comment (Camork) [reply tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java]: In some method we need to return error ` IStatus` and in another situation we need to log error. For example, ` LiferayWorkspaceUtil.addPortalRuntime()` invoked by `InitBundleaction` and `GradleWorkspaceProvider.initBundle()`.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java (hunk 4 above)
Comment (Camork) [reply tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/ProjectCore.java]: I found these code is invoked by no one.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/NewLiferayWorkspaceProjectProvider.java (hunk 6 above)
Comment (Camork) [reply tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/NewLiferayWorkspaceProjectProvider.java]: Just for preciseness, I could't assume project folder name is project name the situation we discussed this pm.

**Pull request closed without merging**